### PR TITLE
Update RegistryError to use top level errors

### PR DESCRIPTION
**Pull request by agunde406**
Replaces the existing error types with InternalError,
InvalidState, ConstraintViolationError, and
ResourceTemporarilyUnavailableError.

Also adds a error that can be used in the REST API routes.
The Registry error can no longer be used in a web::block,
because Box<dyn Error> is guaranteed to be Send. The
new error will conver the RegistryError to either InvalidState,
which is required for returning a BadRequest, and
InternalError that will be logged and InternalServerError
is returned.

Signed-off-by: Andrea Gunderson <agunde@bitwise.io>

### CODE REVIEW

File: libsplinter/src/admin/service/mod.rs [hunk 1]
```diff
@@ -131,10 +131,11 @@ impl AdminKeyVerifier for dyn RegistryReader {
     /// registry and the node has the given key. Otherwise, the key is not permitted.
     fn is_permitted(&self, node_id: &str, key: &[u8]) -> Result<bool, AdminKeyVerifierError> {
         let node_opt = self.fetch_node(node_id).map_err(|err| {
-            AdminKeyVerifierError::new_with_source(
-                &format!("Failed to lookup node '{}' in registry", node_id),
-                Box::new(err),
-            )
+            AdminKeyVerifierError::new(&format!(
```
Comment (vaporos) [inline libsplinter/src/admin/service/mod.rs:134]: Wondering if we can use a pattern that doesn't require this error to be Send, allowing it to keep the new_with_source() method. Maybe handling it at the channel boundary by not sending AdminKeyVerifierError directly, but sending its display and debug instead.

Comment (agunde406) [reply libsplinter/src/admin/service/mod.rs:134]: We should have a larger discussion about this pattern and update this in another PR based on that conversation.

**Review by vaporos — DISMISSED** [review]

File: libsplinter/src/admin/service/mod.rs (hunk 1 above)
Comment (ltseeley) [reply libsplinter/src/admin/service/mod.rs:134]: I think we avoid having to do this in some endpoints by not using `web::block` and instead just converting the HTTP response into a future like `return Box::new(HttpResponse::Ok().finish().into_future())`

File: libsplinter/src/registry/diesel/mod.rs [hunk 2]
```diff
@@ -209,6 +209,7 @@ pub mod tests {
 
         assert_eq!(node, get_node_1());
 
+        #[allow(deprecated)]
```
Comment (ltseeley) [inline libsplinter/src/registry/diesel/mod.rs:212]: This seems like a separate change

**Review by ltseeley — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**